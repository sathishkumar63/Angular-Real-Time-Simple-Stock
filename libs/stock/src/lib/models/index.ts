export * from './stock.constant';
export * from './stock.interface';
export * from './stock.mock.data';
