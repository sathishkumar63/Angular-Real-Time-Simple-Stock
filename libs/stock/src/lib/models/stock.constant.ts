export const stocksSymbol: string[] = [
  'GOOG',
  'GOOGL',
  'MSFT',
  'AAPL',
  'TSLA',
  'AMZN',
  'UBER',
  'DIDI',
  'META',
  'NFLX',
  'CHIK',
  'GLD',
  'IAU',
  'BTC-USD',
  'ETH-USD',
  'DOGE-USD',
  'SHIB-USD',
  'ETH-CAD',
  'USDT-USD',
  'USDC-USD',
];
