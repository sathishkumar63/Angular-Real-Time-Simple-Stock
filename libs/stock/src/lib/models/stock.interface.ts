export interface IStock {
  name?: string;
  symbol: string;
  currency?: string;
  currentPrice: number;
  dayHighPrice?: number;
  dayLowPrice?: number;
  fiftyTwoWeekLow?: number;
  fiftyTwoWeekHigh?: number;
  regularMarketVolume?: number;
  isActive?: boolean;
}
