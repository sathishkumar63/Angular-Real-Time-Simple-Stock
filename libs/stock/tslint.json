{
  "extends": "../../tslint.json",
  "rules": {
    "directive-selector": [true, "attribute", "stock", "camelCase"],
    "component-selector": [true, "element", "stock", "kebab-case"]
  },
  "linterOptions": {
    "exclude": ["!**/*"]
  }
}
