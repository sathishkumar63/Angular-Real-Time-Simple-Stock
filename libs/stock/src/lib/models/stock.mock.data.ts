import { StocksFeatureState, StocksState } from '../+state';
import { IStock } from './stock.interface';

export const mockStocks: IStock[] = [
  {
    name: 'Alphabet',
    symbol: 'GOOG',
    currency: 'USD',
    currentPrice: 2228.55,
    dayHighPrice: 2269.94,
    dayLowPrice: 2217.22,
    fiftyTwoWeekLow: 2044.16,
    fiftyTwoWeekHigh: 3042,
    regularMarketVolume: 1567487,
  },
  {
    name: 'Microsoft',
    symbol: 'MSFT',
    currency: 'USD',
    currentPrice: 252.99,
    dayHighPrice: 260.54,
    dayLowPrice: 252.86,
    fiftyTwoWeekLow: 246.44,
    fiftyTwoWeekHigh: 349.67,
    regularMarketVolume: 31445841,
  },
];

export const mockStocksState: StocksState = {
  data: mockStocks,
  loading: false,
  loaded: true
};

export const mockStocksFeatureState: StocksFeatureState = {
  stocks: mockStocksState
};
