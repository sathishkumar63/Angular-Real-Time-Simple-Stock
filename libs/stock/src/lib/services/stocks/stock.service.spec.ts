import { HttpClientModule } from '@angular/common/http';
import {
  HttpClientTestingModule,
  HttpTestingController,
} from '@angular/common/http/testing';
import { TestBed } from '@angular/core/testing';
import { mockStocks, QUOTE_URL } from '../../models';
import { StockService } from './stock.service';

describe('StockService', () => {
  let service: StockService;
  let mockHttp: HttpTestingController;

  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [HttpClientModule, HttpClientTestingModule],
    });
    service = TestBed.inject(StockService);
    mockHttp = TestBed.inject(HttpTestingController);
  });

  it('should be created', () => {
    expect(service).toBeTruthy();
  });

  describe('getStocks', () => {
    it('should return stocks', () => {
      service
        .getStocks(['tsla'])
        .subscribe((data) => expect(data).toEqual(mockStocks));
      const req = mockHttp.expectOne(
        `https://cors-anywhere.herokuapp.com/${QUOTE_URL}symbols=tsla`
      );
      expect(req.request.method).toEqual('GET');
      req.flush(mockStocks);
      mockHttp.verify();
    });
  });

  describe('formatStocksResponse', () => {
    it('should format and return the stocks', () => {
      // arrange + act
      const result = service.formatStocksResponse(mockStocks);
      // assert
      expect(result).toMatchSnapshot();
    });
  });
});
