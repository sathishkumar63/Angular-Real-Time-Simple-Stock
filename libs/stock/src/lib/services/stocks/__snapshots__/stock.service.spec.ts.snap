// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StockService formatStocksResponse should format and return the stocks 1`] = `
Array [
  Object {
    "currency": "USD",
    "currentPrice": undefined,
    "dayHighPrice": undefined,
    "dayLowPrice": undefined,
    "fiftyTwoWeekHigh": 3042,
    "fiftyTwoWeekLow": 2044.16,
    "name": undefined,
    "regularMarketVolume": 1567487,
    "symbol": "GOOG",
  },
  Object {
    "currency": "USD",
    "currentPrice": undefined,
    "dayHighPrice": undefined,
    "dayLowPrice": undefined,
    "fiftyTwoWeekHigh": 349.67,
    "fiftyTwoWeekLow": 246.44,
    "name": undefined,
    "regularMarketVolume": 31445841,
    "symbol": "MSFT",
  },
]
`;
